### PR TITLE
feat(perf): Add `http_response_code_rate` span metrics function

### PR DESCRIPTION
**Pull request by gggritso**
Preparation for upcoming HTTP module. This lets us have columns like "5XXs" rate in the table.

- adds a new `http_response_count_rate` function, backed by `http_response_code_count` function
- accepts one argument, and does a prefix match so it's possible to get `http_response_code_count(200)` and get the 200s rate _or_ do `http_response_code_count(5)` and get the rate of all 500s

## Questions

I have lots, since this is my first PR of this kind. Beyond the obvious "does any of this make sense":

1. Should I change the existing `http_error_rate` function in any way? I can make it rely on the new `http_response_code_rate` function, or leave it alone, or deprecate it
2. What kind of documentation does this function need?
3. Does the name `http_response_code_rate` make sense? Is this API comfortable?

### CODE REVIEW

File: src/sentry/search/events/datasets/spans_metrics.py [hunk 1]
```diff
@@ -585,6 +618,42 @@ def _resolve_time_spent_percentage(
             alias,
         )
 
+    def _resolve_http_response_code_count(
+        self,
+        args: Mapping[str, str | Column | SelectType | int | float],
+        alias: str | None = None,
+        extra_condition: Function | None = None,
```
Comment (wmak) [inline src/sentry/search/events/datasets/spans_metrics.py:625]: should just remove the extra_condition param

**Comment by wmak** [comment]
> 1. Should I change the existing http_error_rate function in any way? I can make it rely on the new http_response_code_rate function, or leave it alone, or deprecate it

If we're no longer using it on the frontend and this dataset isn't public in the API yet, we should just remove it
If we're still using it, lets comment that its deprecate it and remove it later

> 2. What kind of documentation does this function need?

😬 Documentation on these functions is sparse right now so unless you're feeling particularly keen, can skip this for now

> Does the name http_response_code_rate make sense? Is this API comfortable?

Hmmm I don't have strong feelings on any of the following;
- I sorta wish it was shorter? `http_response_rate` instead
- But longer names self document in a way, in which case i sorta want to include the prefix match in? `http_response_by_prefix_rate`

File: src/sentry/search/events/datasets/spans_metrics.py (hunk 1 above)
Comment (gggritso) [reply src/sentry/search/events/datasets/spans_metrics.py]: Done!

**Comment by gggritso** [comment]
@wmak went with the shorter name! Leaving the existing error rate function alone for now (added comment) since the FE mentions it in a few places, but I can clean that up in the near future

**Review by wmak — APPROVED** [review]

**Pull request merged**